src/com/gh4a/activities/IssueListActivity.java

```diff
@@ -333,11 +324,17 @@ protected boolean fragmentNeedsRefresh(Fragment object) {
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == REQUEST_ISSUE_CREATE) {
-            if (resultCode == Activity.RESULT_OK) {
-                // delay refresh for a bit, as
-                // - the fragment might not be created yet
-                // - Github seems to need some time to reflect new issues in their response
-                mHandler.postDelayed(mRefreshRunnable, 500);
+            if (resultCode == Activity.RESULT_OK && data.hasExtra("issue")) {
+                final Issue issue = (Issue) data.getSerializableExtra("issue");
+                // Delay click as the fragment might not be created yet
+                mHandler.post(new Runnable() {
```

Comment:
This delay seems a little unsafe. if mOpenFragment is null, we probably rather need to memorize the passed issue and carry out the onItemClick when we actually have the fragment reference. Or - probably preferable - just do the `startActivityForResult(IssueActivity.makeIntent(this, mRepoOwner, mRepoName, issue.getNumber())` ourselves.